### PR TITLE
Implemented functions for skew-symmetric form of cross product

### PR DESCRIPTION
**Pull request by evbernardes**
#### References to other Issues or PRs
Added `hat` and `vee` convenient functions for matrices.
`hat` transforms a `3 x 1` Matrix into a `3 x 3` skew-symmetric matrix, and `vee` does the inverse operation.

I'm open to changing the names of the functions to something more descriptive if needed, these are just the names I'm more used to use when studying Lie theory, as seen for example in [this library](https://github.com/artivis/manif).

#### Brief description of what is fixed or changed
The cross product $\pmb v_1 \times \pmb v_2$ can be expressed instead by $\widehat{\pmb v}_1  \pmb v_2$, where $\widehat{\pmb v}_1$ is a `3 x 3` skew symmetric matrix with the property that $(\widehat{\pmb v}_1)^T = -\widehat{\pmb v}_1$.

For convenience, the inverse operation is also implemented.

#### Release Notes

* matrices
  * Implemented `hat` and `vee` operations for matrices.

### CODE REVIEW

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -1260,6 +1262,65 @@ def cross(self, b):
...
+        cross
+        hat
+        multiply
+        multiply_elementwise
+        """
+
+        if not (self.shape == (3, 3)):
+            raise ShapeError("Dimensions incorrect, expected (3, 3), got " +
+                             str(self.shape))
+        elif self.T != -self:
```
Comment (sylee957) [inline sympy/matrices/matrices.py:1316]:
```suggestion
        elif not self.is_anti_symmetric():
```

File: sympy/matrices/matrices.py [hunk 2]
```diff
@@ -1260,6 +1262,57 @@ def cross(self, b):
...
+        ========
+
+        dot
+        cross
+        vee
+        multiply
+        multiply_elementwise
+        """
+
+        if not (self.shape == (3, 1)):
```
Comment (sylee957) [inline sympy/matrices/matrices.py:1280]:
```suggestion
        if self.shape != (3, 1):
```

File: sympy/matrices/matrices.py [hunk 3]
```diff
@@ -1260,6 +1262,57 @@ def cross(self, b):
...
+        ========
+
+        dot
+        cross
+        hat
+        multiply
+        multiply_elementwise
+        """
+
+        if not (self.shape == (3, 3)):
```
Comment (sylee957) [inline sympy/matrices/matrices.py:1305]:
```suggestion
        if self.shape != (3, 3):
```

**Comment by sylee957** [comment]
The documentation may be too small

https://output.circle-artifacts.com/output/job/193e6766-24c1-4216-988c-cc889f8db09e/artifacts/0/doc/_build/html/modules/matrices/matrices.html#sympy.matrices.matrices.MatrixBase.hat

I think that some examples should be added

**Comment by evbernardes** [comment]
> I think that some examples should be added

Fair, just added some!

File: sympy/matrices/matrices.py [hunk 4]
```diff
@@ -1260,6 +1262,145 @@ def cross(self, b):
...
+        Matrix([
+        [                      0],
+        [                      0],
+        [Derivative(theta(t), t)]])
+
+
+
+
+
+
```
Comment (sylee957) [inline sympy/matrices/matrices.py:1382]:
```suggestion
```

Comment (evbernardes) [reply sympy/matrices/matrices.py]: I had just noticed that, already fixed!

**Comment by evbernardes** [comment]
Thanks for merging @sylee957, there was still a test that was not passing though. 
Was it related to something else?

**Comment by sylee957** [comment]
There's a upstream bug with python 3.12 in #24802

**Pull request merged**